Plugins/ch.zhaw.graphcut/src/internal/lib/GraphCut3D/ImageMultiLabelGridCutFilter.hxx

```diff
@@ -93,37 +91,43 @@ namespace itk {
 		unsigned int nGraphNodes(1);
 
         unsigned int dim(graphSize.GetSizeDimension());
-		for (int iSize = 0; iSize < dim; ++iSize) {
+		for (unsigned int iSize = 0; iSize < dim; ++iSize) {
 			nGraphNodes *= graphSize[iSize];
 		}
 
         WeightType * dataCosts = new WeightType[nGraphNodes * nLabels];
-        for (int iDatacost = 0; iDatacost < nGraphNodes * nLabels; ++iDatacost) {
-            if (std::numeric_limits<WeightType>::max()<std::numeric_limits<float>::max())
-                dataCosts[iDatacost] = std::numeric_limits<WeightType >::max() - 1;
-            else
-                dataCosts[iDatacost] = 1000;
+
+        WeightType dataCostTmp;
+        if (std::numeric_limits<WeightType>::max()<std::numeric_limits<float>::max())
+            dataCostTmp = std::numeric_limits<WeightType >::max() - 1;
+        else
+            dataCostTmp = 1000;
+
+        for (unsigned int iDatacost = 0; iDatacost < nGraphNodes * nLabels; ++iDatacost) {
+            dataCosts[iDatacost] = dataCostTmp;
         }
-        for (int iLabel = 0; iLabel < nLabels; ++iLabel) {
+        for (unsigned int iLabel = 0; iLabel < nLabels; ++iLabel) {
             auto indexArray = labels[iLabel]; // an array of 3d image coordinates for the iLabel
-            for (int iVoxel = 0; iVoxel < indexArray.size(); ++iVoxel) {
+            for (unsigned int iVoxel = 0; iVoxel < indexArray.size(); ++iVoxel) {
                 assert(images.multiLabel->ComputeOffset(indexArray[iVoxel]) == this->ConvertIndexToVertexDescriptor(indexArray[iVoxel], images.multiLabel->GetLargestPossibleRegion()));
                 dataCosts[images.multiLabel->ComputeOffset(indexArray[iVoxel]) * nLabels + iLabel] =  0;
             }
         }
 
-        WeightType** smoothnessCosts = new WeightType*[nGraphNodes * dim];
+        WeightType** smoothnessCosts = new WeightType*[nGraphNodes * neighbors.size()];
+        // store the weight in a std vector because the pointers in the smoothnessCosts array are not released in the gridCut library
+        mWeights.resize(nGraphNodes * neighbors.size());
         unsigned int iVoxel(0);
 		for (iterator.GoToBegin(); !iterator.IsAtEnd(); ++iterator, ++iVoxel) {
```

Comment:
Indentation is kinda messed up in this whole file. Can you maybe run it through an auto formatter?